python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -1,169 +1,78 @@
-        if isinstance(meta, dict):
-            self.input_types = meta["input_types"]
-            self.input_lengths = meta["input_lengths"]
-            return
-        buffers = meta
-        self.input_types = []
-        self.input_lengths = []
-        if buffers.points is not None:
-            self.input_types.extend(repeat(0, len(buffers.points)))
-            self.input_lengths.extend(repeat(1, len(buffers.points)))
-        if buffers.multipoints is not None:
-            self.input_types.extend(repeat(1, len(buffers.multipoints)))
-            self.input_lengths.extend(repeat(1, len(buffers.multipoints)))
-        if buffers.lines is not None:
-            for index in range(len(buffers.lines.mlines) - 1):
-                line_len = (
-                    buffers.lines.mlines[index + 1]
-                    - buffers.lines.mlines[index]
-                )
-                if line_len > 1:
-                    self.input_types.extend([3])
-                    self.input_lengths.extend([line_len])
-                else:
-                    self.input_types.extend([2])
-                    self.input_lengths.extend([1])
-        if buffers.polygons is not None:
-            for index in range(len(buffers.polygons.mpolys) - 1):
-                poly_len = (
-                    buffers.polygons.mpolys[index + 1]
-                    - buffers.polygons.mpolys[index]
-                )
-                if poly_len > 1:
-                    self.input_types.extend([5])
-                    self.input_lengths.extend([poly_len])
-                else:
-                    self.input_types.extend([4])
-                    self.input_lengths.extend([1])
-        self.input_types = pa.array(self.input_types, type=pa.int8())
-        self.input_lengths = pa.array(self.input_lengths).cast(pa.int32())
-
-    def copy(self):
-        return self.__class__(
-            {
-                "input_types": pa.Int8Array.from_buffers(
-                    self.input_types.type,
-                    len(self.input_types),
-                    self.input_types.buffers(),
-                ),
-                "input_lengths": pa.Int32Array.from_buffers(
-                    self.input_lengths.type,
-                    len(self.input_lengths),
-                    self.input_lengths.buffers(),
-                ),
-            }
-        )
-
-
 class GeoColumn(NumericalColumn):
     """
     Parameters
     ----------
-    data : A GeoArrowBuffers object
+    data : A tuple of four cudf.ListSeries
     meta : A GeoMeta object (optional)
 
     Notes
     -----
     The GeoColumn class subclasses `NumericalColumn`. Combined with
-    `_copy_type_metadata`, this assures support for existing cudf algorithms.
+    `_copy_type_metadata`, this assures support for sort, groupby,
+    and potential other `cudf` algorithms.
     """
 
     def __init__(
         self,
-        data: GeoArrowBuffers,
+        data: Tuple,
```

Comment:
I could take a Tuple of buffers instead of a tuple of Series? Then I'd have to construct the Series that are used internally here. I appreciate the comment.